homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,255 @@
+    "activityStressPercentage": ["Activity Stress Percentage", "%", "mdi:flash-alert"],
+    "uncategorizedStressPercentage": [
+        "Uncat. Stress Percentage",
+        "%",
+        "mdi:flash-alert",
+    ],
+    "lowStressPercentage": ["Low Stress Percentage", "%", "mdi:flash-alert"],
+    "mediumStressPercentage": ["Medium Stress Percentage", "%", "mdi:flash-alert"],
+    "highStressPercentage": ["High Stress Percentage", "%", "mdi:flash-alert"],
+    "moderateIntensityMinutes": ["Moderate Intensity", "minutes", "mdi:flash-alert"],
+    "vigorousIntensityMinutes": ["Vigorous Intensity", "minutes", "mdi:run-fast"],
+    "intensityMinutesGoal": ["Intensity Goal", "minutes", "mdi:run-fast"],
+    "bodyBatteryChargedValue": [
+        "Body Battery Charged",
+        "%",
+        "mdi:battery-charging-100",
+    ],
+    "bodyBatteryDrainedValue": [
+        "Body Battery Drained",
+        "%",
+        "mdi:battery-alert-variant-outline",
+    ],
+    "bodyBatteryHighestValue": ["Body Battery Highest", "%", "mdi:battery-heart"],
+    "bodyBatteryLowestValue": ["Body Battery Lowest", "%", "mdi:battery-heart-outline"],
+    "bodyBatteryMostRecentValue": [
+        "Body Battery Most Recent",
+        "%",
+        "mdi:battery-positive",
+    ],
+    "averageSpo2": ["Average SPO2", "%", "mdi:diabetes"],
+    "lowestSpo2": ["Lowest SPO2", "%", "mdi:diabetes"],
+    "latestSpo2": ["Latest SPO2", "%", "mdi:diabetes"],
+    "latestSpo2ReadingTimeLocal": ["Latest SPO2 Time", "", "mdi:diabetes"],
+    "averageMonitoringEnvironmentAltitude": [
+        "Average Altitude",
+        "%",
+        "mdi:image-filter-hdr",
+    ],
+    "durationInMilliseconds": ["Duration", "ms", "mdi:progress-clock"],
+    "highestRespirationValue": ["Highest Respiration", "brpm", "mdi:progress-clock"],
+    "lowestRespirationValue": ["Lowest Respiration", "brpm", "mdi:progress-clock"],
+    "latestRespirationValue": ["Latest Respiration", "brpm", "mdi:progress-clock"],
+    "latestRespirationTimeGMT": ["Latest Respiration Update", "", "mdi:progress-clock"],
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+GARMIN_DEFAULT_CONDITIONS = ["totalSteps"]
+ATTRIBUTION = "Data provided by garmin.com"
+DEFAULT_NAME = "Garmin"
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_EMAIL): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(
+            CONF_MONITORED_CONDITIONS, default=GARMIN_DEFAULT_CONDITIONS
+        ): vol.All(cv.ensure_list, [vol.In(GARMIN_CONDITIONS_LIST)]),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Garmin Connect component."""
+
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+    prefix_name = config.get(CONF_NAME)
+
+    try:
+        garmin_client = garminconnect.Garmin(email, password)
```

Comment:
Do they have OAuth2 available ?

Reply:
This integration is using the JSON rest API calls as used by the Garmin Connect website, and doesn't support OAuth as far as I could see.
The official Garmin API does, but I believe its a paid service. (I think with a $5000 fee)
I will have another look.